workspace/path_type.go

```diff
@@ -0,0 +1,61 @@
+package workspace
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// PathType is either a path to a dir or file, or the name of an exercise.
+type PathType int
+
+const (
+	// TypeExerciseID is the name of an exercise.
+	TypeExerciseID PathType = iota
+	// TypeDir is a relative or absolute path to a directory.
+	TypeDir
+	// TypeFile is a relative or absolute path to a file.
+	TypeFile
+)
+
+// DetectPathType determines whether the given path is a directory, a file, or the name of an exercise.
+func DetectPathType(path string) (PathType, error) {
+	// If it's not an absolute path, make it one.
+	if !filepath.IsAbs(path) {
+		var err error
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return 0, err
```

Comment:
This is a little confusing seeing as TypeExerciseID is also 0. What do you think about starting the constants at 1 or maybe returning a nagative number.